pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,832 @@
+                [float.fromhex(i) for i in iv_curve["i_gold"]])
+
+    return gold_dataset_converted
+
+
+def pretty_print_gold_dataset(gold_dataset):
+    """
+    Create human-readable gold dataset.
+
+    From a gold dataset in the lossless, machine-interoperable format, create
+    and return the dataset as a human-readale, comma-separated-value (csv)
+    string.
+
+    Parameters
+    ----------
+    gold_dataset : dict
+
+    Returns
+    -------
+    gold_dataset_csv : str
+
+    """
+
+    # Convert gold dataset from hex
+    gold_dataset_converted = convert_gold_dataset(gold_dataset)
+
+    num_pts = len(
+        gold_dataset_converted["devices"][0]["iv_curves"][0]["v_gold"])
+
+    # Initialize output
+    gold_dataset_csv = gold_dataset_converted["timestamp_utc_iso"] + \
+        ", " + gold_dataset_converted["dataset_version"] + "\n\n"
+    gold_dataset_csv = gold_dataset_csv + "G (W/m^2), T (degC)" + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        "r_sh (Ohm), r_s (Ohm), nNsVth (V), i_0 (A), i_l (A)" + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        ", ".join(["v_gold_" + str(idx) +
+                   " (V)" for idx in range(num_pts)]) + "\n"
+    gold_dataset_csv = gold_dataset_csv + \
+        ", ".join(["i_gold_" + str(idx) +
+                   " (A)" for idx in range(num_pts)]) + "\n"
+    gold_dataset_csv = gold_dataset_csv + "\n\n"
+
+    for device in gold_dataset_converted["devices"]:
+        for iv_curve in device["iv_curves"]:
+            # Load G, T, and 5 single-diode model parameters for each I-V curve
+            # Floats are converted from lossless, interchangeable hex format
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, [iv_curve["G"], iv_curve["T"]])) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, [
+                    iv_curve["r_sh"], iv_curve["r_s"], iv_curve["nNsVth"],
+                    iv_curve["i_0"], iv_curve["i_l"]
+                ])) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, iv_curve["v_gold"].tolist())) + "\n"
+            gold_dataset_csv = gold_dataset_csv + ", ".join(
+                map(str, iv_curve["i_gold"].tolist())) + "\n"
+
+    return gold_dataset_csv
+
+
+def save_gold_dataset(gold_dataset,
+                      json_rel_path="sdm.json",
+                      csv_rel_path="sdm.csv"):
+    """
+    Save gold dataset to file in two possible formats.
+
+    The gold dataset dict is saved in a lossless, machine-interoperable json
+    file and in a lossy, but human-readable, csv file.
+
+    Relative file paths are used, which are relative to the current directory.
+    Defaults are customizable. Set a file path to None to skip saving that
+    file.
+
+    Parameters
+    ----------
+    gold_dataset : dict
+
+    json_rel_path : str
+        Pass ``None`` to skip writing this file.
```

Comment:
I'd probably go with a default ``None``. But I'll let it go if no one else agrees with me.